diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,235 @@
+import org.cqfn.diktat.ruleset.constants.Warnings.WHITESPACE_IN_COMMENT
+import org.cqfn.diktat.ruleset.rules.getDiktatConfigRules
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.getFirstChildWithType
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+
+
+/**
+ * This class handles rule 2.6
+ * Part 1:
+ * there must be 1 space between the comment character and the content of the comment;
+ * there must be a newline between a Kdoc and the previous code above.
+ * No need to add a blank line before a first comment in this particular name space (code block), for example between function declaration and first comment in a function body.
+ *
+ * Part 2:
+ * Leave one single space between the comment on the right side of the code and the code.
+ * comments in if-else
+ */
+class CommentsFormatting : Rule("kdoc-comments-codeblocks-formatting") {
+
+    companion object {
+        private const val MAX_SPACES = 1
+        private const val APPROPRIATE_COMMENT_SPACES = 3
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+
+        configRules = params.getDiktatConfigRules()
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        val configuration = KdocCodeBlocksFormattingConfiguration(
+                configRules.getRuleConfig(SPACE_BETWEEN_COMMENT_AND_CODE)?.configuration ?: mapOf())
+
+        when (node.elementType) {
+            IF -> {
+                handleIfElse(node)
+            }
+            EOL_COMMENT -> {
+                handleEolComments(node, configuration)
+            }
+            BLOCK_COMMENT -> {
+                handleBlockComments(node, configuration)
+            }
+
+            KDOC -> {
+                handleKdocComments(node)
+            }
+        }
+    }
+
+    private fun handleKdocComments(node: ASTNode) {
+        if (node.treeParent.treeParent.elementType == BLOCK) {
+            checkBlockComments(node.treeParent) // node.treeParent is a Property.
+        } else if (node.treeParent.elementType != IF){
+            checkClassComment(node)
+        }
+    }
+
+    private fun handleEolComments(node: ASTNode, configuration: KdocCodeBlocksFormattingConfiguration) {
+        checkSpaceBetweenPropertyAndComment(node, configuration)
+
+        if (node.treeParent.elementType == BLOCK && node.treeParent.lastChildNode != node) {
```

Comment:
```suggestion
        if (node.treeParent.elementType == BLOCK && node.treeNext != null) {
```
If you want to check if the node is the last child.